thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -740,11 +740,14 @@ class CIndexReadSlaveActivity : public CIndexReadSlaveBase
             steppingMeta.setExtra(steppedExtra);
 
         // NB: setup remoteLimit before base start() call which if parts remote, will use remoteLimit
-        if (choosenLimit)
+        if (!helper->transformMayFilter() && !helper->hasMatchFilter())
         {
-            remoteLimit = choosenLimit;
-            if (!helper->transformMayFilter() && (RCMAX != keyedLimit) && (keyedLimit+1 < remoteLimit))
-                remoteLimit = keyedLimit+1; // 1 more to ensure triggered when received back.
+            if (choosenLimit)
+            {
+                remoteLimit = choosenLimit;
+                if ((RCMAX != keyedLimit) && (keyedLimit+1 < remoteLimit))
```

Comment:
Feels like we should be able to use keyedLimit regardless of whether choosen is set and regardless of whether transformMayFilter or hasMatchFilter

Reply:
I agree, but I hesitate to change without Jake to review.  Should it be split into a separate jira @richardkchapman ?